File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/booking/InvoiceDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto.booking;

import at.ac.tuwien.sepm.groupphase.backend.entity.InvoiceType;
import java.math.BigDecimal;
import java.sql.Timestamp;
import java.util.List;

public record InvoiceDto(
    Long invoiceNumber,
    String companyLocation,
    String taxNumber,
    Timestamp invoiceDate,
    Long bookingId,
    InvoiceType invoiceType,
    Long cancelledInvoiceId,
    List<InvoiceLine> lines,
    BigDecimal netSum,
    BigDecimal vatAmount,
    BigDecimal grossSum,
    String footerText) {

  public record InvoiceLine(
      String eventTitle,
      Integer quantity,
      String location,
      String room,
      String sector,
      String type,
      BigDecimal pricePerUnit) {}
}

File: frontend/src/app/dtos/booking/invoice.dto.ts
export interface InvoiceLineDto {
    readonly eventTitle: string;
    readonly quantity: number;
    readonly location: string;
    readonly room: string;
    readonly sector: string;
    readonly type: 'seating' | 'standing';
    readonly pricePerUnit: number;
}

export interface InvoiceDto {
    readonly invoiceNumber: number;
    readonly companyLocation: string;
    readonly taxNumber: string;
    readonly invoiceDate: Date;
    readonly bookingId: number;
    readonly invoiceType: 'purchase' | 'cancellation';
    readonly cancelledInvoiceId?: number;
    readonly netSum: number;
    readonly vatAmount: number;
    readonly grossSum: number;
    readonly footerText: string;
    readonly lines: InvoiceLineDto[];
}
